components/chainhook-cli/src/service/tests/mod.rs

```diff
@@ -0,0 +1,334 @@
+#[test_case(json!({"http_post": {"url": "http://localhost:1234", "authorization_header": "Bearer FYRPnz2KHj6HueFmaJ8GGD3YMbirEFfh"}}) ; "with http_post action")]
+#[test_case(json!({"file_append": {"path": "./path"}}) ; "with file_append action")]
+#[tokio::test]
+async fn it_handles_bitcoin_then_that_predicates(then_that: JsonValue) {
+    let predicate = build_bitcoin_payload(None, None, Some(then_that), None);
+    match test_register_predicate(predicate).await {
+        Ok(_) => {}
+        Err((e, shutdown)) => {
+            shutdown.notify();
+            panic!("{e}");
+        }
+    }
+}
+
+#[test_case(json!({"blocks": [0, 1, 2],"start_block": 0,"end_block": 0,"expire_after_occurrence": 0,"include_proof": true,"include_inputs": true,"include_outputs": true,"include_witness": true}) ; "all filters")]
+#[test_case(json!({"blocks": [0, 1, 2]}) ; "blocks filter")]
+#[test_case(json!({"start_block": 0}) ; "start_block filter")]
+#[test_case(json!({"end_block": 0}) ; "end_block filter")]
+#[test_case(json!({"expire_after_occurrence": 0}) ; "expire_after_occurrence filter")]
+#[test_case(json!({"include_proof": true}) ; "include_proof filter")]
+#[test_case(json!({"include_inputs": true}) ; "include_inputs filter")]
+#[test_case(json!({"include_outputs": true}) ; "include_outputs filter")]
+#[test_case(json!({"include_witness": true}) ; "include_witness filter")]
+#[tokio::test]
+async fn it_handles_bitcoin_predicates_with_filters(filters: JsonValue) {
+    let predicate = build_bitcoin_payload(None, None, None, Some(filters));
+    match test_register_predicate(predicate).await {
+        Ok(_) => {}
+        Err((e, shutdown)) => {
+            shutdown.notify();
+            panic!("{e}");
+        }
+    }
+}
+
+#[test_case("mainnet" ; "mainnet")]
+#[test_case("testnet" ; "testnet")]
+#[test_case("devnet" ; "devnet")]
+#[test_case("simnet" ; "simnet")]
+#[tokio::test]
+async fn it_handles_stacks_predicates_with_network(network: &str) {
+    let predicate = build_stacks_payload(Some(network), None, None, None);
+    match test_register_predicate(predicate).await {
+        Ok(_) => {}
+        Err((e, shutdown)) => {
+            shutdown.notify();
+            panic!("{e}");
+        }
+    }
+}
+
+#[test_case(json!({"scope":"block_height", "equals": 100}); "with scope block_height equals match")]
+#[test_case(json!({"scope":"block_height", "higher_than": 100}); "with scope block_height higher_than match")]
+#[test_case(json!({"scope":"block_height", "lower_than": 100}); "with scope block_height lower_than match")]
+#[test_case(json!({"scope":"block_height", "between": [100,102]}); "with scope block_height between match")]
+#[test_case(json!({"scope":"contract_deployment", "deployer": "ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM"}); "with scope contract_deployment type deployer")]
+// #[test_case(json!({"scope":"contract_deployment", "implement_sip09": 100}); "with scope contract_deployment type implement_sip09")]
```

Comment:
@lgalabru I couldn't get these two `contract_deployment` predicates to work. What should this look like?

Reply:
These 2 predicates are not ready, we're missing the bulk of their implementation (pretty sure that they've been marked as `coming soon` in our docs).